llvm/tools/llvm-gsymutil/llvm-gsymutil.cpp

```diff
@@ -515,10 +520,31 @@ int llvm_gsymutil_main(int argc, char **argv, const llvm::ToolContext &) {
     // Call error() if we have an error and it will exit with a status of 1
     if (auto Err = convertFileToGSYM(Aggregation))
       error("DWARF conversion failed: ", std::move(Err));
+
     // Report the errors from aggregator:
     Aggregation.EnumerateResults([&](StringRef category, unsigned count) {
       OS << category << " occurred " << count << " time(s)\n";
     });
+    if (!AggregateJsonFile.empty()) {
+      std::error_code EC;
+      raw_fd_ostream JsonStream(AggregateJsonFile, EC, sys::fs::OF_Text);
+      if (EC) {
+        OS << "error opening aggregate error json file '" << AggregateJsonFile
+           << "' for writing: " << EC.message() << '\n';
+        return 1;
+      }
+      JsonStream << "{\"errors\":[\n";
+      bool prev = false;
+      Aggregation.EnumerateResults([&](StringRef category, unsigned count) {
+        if (prev)
+          JsonStream << ",\n";
+        JsonStream << "{\"category\":\"";
+        llvm::printEscapedString(category, JsonStream);
+        JsonStream << "\",\"count\":" << count << "}";
+        prev = true;
+      });
+      JsonStream << "\n]}\n";
```

Comment:
Instead of making manual JSON, I would use the llvm JSON objects.  There are many examples in lldb/tools/lldb-dap/lldb-dap.cpp or in lldb/tools/lldb-dap/JSONUtils.cpp.

We should probably agree on the format first. I was thinking something like a top level dictionary with a value called "errors" that contains a dictionary of `"<category>" : <object>`:
```
{ "error-categories":  
  {
    "Overlapping address ranges":  {"count": 12 },
    "Invalid file index":  {"count": 12 },
  }
}
```
This way if we eventually add information per category, like say a vector of DIE offsets that show where this issue is in the DWARF, can add it without any changes to the format:
```
{ "error-categories":  
  {
    "Overlapping address ranges":  {"count": 12, "dies": [12312312, 223423423] },
    "Invalid file index":  {"count": 32, , "dies": [443423432, 234234323]},
  }
}
```

Reply:
To create this kind of thing with objects the JSON code would look like:
```
llvm::json::Object category_dict;
Aggregation.EnumerateResults([&](StringRef category, unsigned count) {
  llvm::json::Object category_info;
  category_info.try_emplace("count", count);
  category_dict.try_emplace(category, category_info);
});
llvm::json::Object json_output;
json_output.try_emplace("error-categories", category_dict);
// Convert JSON objects to a string
std::string s;
llvm::raw_string_ostream strm(s);
strm << json_output;
```

Reply:
Even if we use llvm::json objects to create the JSON we need to make sure we emit valid JSON. Here if the category strings have special JSON reserved characters, it would create invalid JSON, so we need to make sure the category strings have special characters, like '"' characters, that it works correctly.

Reply:
I believe that if we use the llvm::json objects this will all be taken care of, but we should verify. If we don't the current code would end up creating invalid JSON if the category string had a '"'...

Reply:
printEscapedString handles double-quotes, but it's still probably better to just do it using JSON objects explicitly...